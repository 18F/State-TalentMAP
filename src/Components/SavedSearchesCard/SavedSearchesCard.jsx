import React from 'react';
import PropTypes from 'prop-types';
import { SAVED_SEARCH_PARENT_OBJECT } from '../../Constants/PropTypes';

const SavedSearchesList = ({ savedSearch, goToSavedSearch, deleteSearch }) => (
  <div className="usa-grid-full saved-search-card" key={savedSearch.id}>
    <div className="usa-grid-full">
      <div className="usa-width-one-whole saved-search-card-name">
        Name: {savedSearch.name}
      </div>
    </div>
    <div className="usa-width-one-whole">
      <button onClick={() => goToSavedSearch(savedSearch)}>View Results</button>
      <button
        className="usa-button-secondary"
        onClick={() => deleteSearch(savedSearch.id)}
      >
        Delete
      </button>
    </div>
  </div>
);

SavedSearchesList.propTypes = {
  savedSearch: SAVED_SEARCH_PARENT_OBJECT.isRequired,
  goToSavedSearch: PropTypes.func.isRequired,
  deleteSearch: PropTypes.func.isRequired,
};

export default SavedSearchesList;
