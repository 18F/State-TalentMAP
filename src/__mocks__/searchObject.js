const searchObject = {
  id: 1,
  name: 'search',
};

export default searchObject;
