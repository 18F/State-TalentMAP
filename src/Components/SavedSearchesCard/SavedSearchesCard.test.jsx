import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import sinon from 'sinon';
import SavedSearchesCard from './SavedSearchesCard';
import searchObject from '../../__mocks__/searchObject';

describe('SavedSearchesCardComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(
      <SavedSearchesCard
        savedSearch={searchObject}
        goToSavedSearch={() => {}}
        deleteSearch={() => {}}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('can receive props', () => {
    const wrapper = shallow(
      <SavedSearchesCard
        savedSearch={searchObject}
        goToSavedSearch={() => {}}
        deleteSearch={() => {}}
      />,
    );
    expect(wrapper.instance().props.savedSearch.id).toBe(searchObject.id);
  });

  it('can call functions on button click', () => {
    const goToSpy = sinon.spy();
    const deleteSpy = sinon.spy();
    const wrapper = shallow(
      <SavedSearchesCard
        savedSearch={searchObject}
        goToSavedSearch={goToSpy}
        deleteSearch={deleteSpy}
      />,
    );
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(1).simulate('click');
    sinon.assert.calledOnce(goToSpy);
    sinon.assert.calledOnce(deleteSpy);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <SavedSearchesCard
        savedSearch={searchObject}
        goToSavedSearch={() => {}}
        deleteSearch={() => {}}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
