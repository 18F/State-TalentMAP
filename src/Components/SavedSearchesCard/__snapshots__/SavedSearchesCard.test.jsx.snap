// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedSearchesCardComponent matches snapshot 1`] = `
<div
  className="usa-grid-full saved-search-card"
>
  <div
    className="usa-grid-full"
  >
    <div
      className="usa-width-one-whole saved-search-card-name"
    >
      Name: 
      search
    </div>
  </div>
  <div
    className="usa-width-one-whole"
  >
    <button
      onClick={[Function]}
    >
      View Results
    </button>
    <button
      className="usa-button-secondary"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
